### ISSUE TITLE
DB out of memory

### ISSUE THREAD

**Issue by Nemi5150** [JWardee/wp-mail-catcher#151]
## Expected Behaviour

When your mail cleanup function runs it uses up all available memory

## Current Behaviour

When the cleanup function runs it fetches all emails, including html text, frmo the DB. This is just way too much at one time. One cleanup is using more than 256MB or ram and gets terminated. 

## My Environment
- WordPress version: X
- PHP version: X
- Plugins I have installed: X, Y, Z

## Steps to Reproduce

1.
2.
3.
4.

## Possible Solution

Ok, I looked at this and your cleanup query gets all email messages at one time and iterates over them. This is just way too much memory. Your cleanup is trying to allocate more than 256MB of memory for just the one php process. 

There is no reason to get all the html text of the email messages just to delete them. This is very inefficient. I have written a new function that is an order of magnitude more efficient. 

In `Logs.php` add the following function:
```
    static public function deleteOlderThan($timeInterval)
    {
        if (empty($timeInterval)) {
            return;
        }

        global $wpdb;

        $timestamp = time() - $timeInterval;

        $sql = $wpdb->prepare("DELETE FROM " . $wpdb->prefix . GeneralHelper::$tableName . "
                      WHERE time <= %d", $timestamp);

        $wpdb->query($sql);
    }
```

and in `ExpiredLogManager.php` change the removeExpiredLogs() function to this:
```
    public static function removeExpiredLogs($timeInterval = null)
    {
//        $idsToRemove = [];
        $interval = $timeInterval == null ?  Settings::get('timescale') : $timeInterval;

//        foreach (Logs::get(['ignore_cache' => true]) as $log) {
//            if ((time() - $log['timestamp']) >= $interval) {
//                $idsToRemove[] = $log['id'];
//            }
//        }

        Logs::deleteOlderThan($interval);
    }

```

**Comment by JWardee** [comment]
You're right that by default when calling the log model it will include the html content by default, I'll shuffle this around so it doesn't pull that column in by default. This will help in a lot of other areas where the logs are queried. 

Thanks for the proposed solution, I'll look at integrating this in the next release.

**Comment by JWardee** [comment]
Implemented in 9a4ae07
